service/src/test/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceIverksettTest.kt

```diff
@@ -62,232 +46,141 @@ internal class SøknadsbehandlingServiceIverksettTest {
 
     @Test
     fun `svarer med feil dersom vi ikke finner behandling`() {
-        val behandlingId = avslagTilAttestering.id
-
-        val søknadsbehandlingRepoMock = mock<SøknadsbehandlingRepo> {
-            on { hent(any()) } doReturn null
-        }
-
-        val ferdigstillVedtakServiceMock = mock<FerdigstillVedtakService>()
-
         val serviceAndMocks = SøknadsbehandlingServiceAndMocks(
-            søknadsbehandlingRepo = søknadsbehandlingRepoMock,
+            søknadsbehandlingRepo = mock {
+                on { hent(any()) } doReturn null
+            },
         )
 
         val response = serviceAndMocks.søknadsbehandlingService.iverksett(
             SøknadsbehandlingService.IverksettRequest(
-                behandlingId,
+                avslagTilAttestering.id,
                 Attestering.Iverksatt(attestant, fixedTidspunkt),
             ),
         )
 
         response shouldBe KunneIkkeIverksette.FantIkkeBehandling.left()
```

Comment:
Imho kunne jeg godt tenkt meg at vi droppet å returnere `left` for caser som helt åpenbart er ute på en snurr. Ikke fordi det elmiminerer behovet for test, men fordi at alle `left` typene er dependency-magneter som må håndteres av alle konsumentene.

Reply:
Dette er en større diskusjon om hvordan vi skal skrive koden vår i så fall 😛  Men enig i at arrow i sin helhet kanskje gir mer kode, men føler at en halvveis tilnærming mellom feks `throw exceptions` og arrow ikke er heldig. Man bør generelt bestemme seg for èn av de.

Reply:
Dette eksempelet går jo under input-validering og er ofte veldig rett fram left's, som ikke må wrappes/oversettes til andre lefts. Vi måtte jo uansett holdt oversikt over hva slags feil dette var og mappet det til riktig http-status og json i web-laget (og forenklinger på dette vil gjelde likt for begge implementasjonsvariantene). Så i dette tilfellet ville det ikke engang vært en kodelinjebesparelse.

Et tilfelle der vi ikke opererer med lefts er connection- og nettverksfeil i forbindelse med databasen.